src/ReactiveDomain.Testing/EventStore/StreamReaderTests.cs

```diff
@@ -75,8 +76,8 @@ public void can_read_stream_forward()
             foreach (var conn in _stores)
             {
                 _count = 0;
-                var reader = new StreamReader("TestReader", conn, _streamNameBuilder, _serializer);
-                reader.EventStream.Subscribe<Event>(this);
+                var reader = new StreamReader("TestReader", conn, _streamNameBuilder, _serializer, evt => { if (evt is Event @event) { this.Handle(@event); } } );
```

Comment:
Why not do this instead?
```
var reader = new StreamReader("TestReader", conn, _streamNameBuilder, _serializer, this.Handle );
```
Does this test need to explicitly exclude messages that aren't of type `Event`? The same thing occurs in a few other tests here.

Reply:
The test fixture handles Event, and the previous implementation would have silently removed all non Event messages, i.e. ResolvedEvents implementing IMessage but not derived from Event. This preserves the same behavior . We can review to see if it is strictly required not that the core behavior has been confirmed to be correct.